### PR TITLE
Feat/user details

### PR DESCRIPTION
**Pull request by MUTABAZI-D**
Hey Boss @kzngoga i have  just just worked user details page , please review it and make necessary comments

### CODE REVIEW

**Review by kzngoga — CHANGES_REQUESTED** [review]
@MUTABAZI-D I didn't review much, but just encountered some unclear implementations & bugs!

- Pls, why did you add the **redux-persist** package?
- Also I keep getting this error when accessing the **users page** https://prnt.sc/YpFcJF2KKPos, not sure why!
- Another thing, is that I deleted all the records in **db.json** but I can still see these workouts: https://prnt.sc/wi56FBMqIJyz
- I think the **redux-persist** package is not needed in any of the usecases in this exercise

File: package.json [hunk 1]
```diff
@@ -28,7 +28,8 @@
     "react-hook-form": "^7.51.5",
     "react-redux": "^9.1.0",
     "react-router-dom": "^6.21.3",
-    "react-toastify": "^10.0.5"
+    "react-toastify": "^10.0.5",
+    "redux-persist": "^6.0.0"
```
Comment (kzngoga) [inline package.json]: Remove this package from **package.json**

File: src/components/users/UserActionMenu.jsx [hunk 2]
```diff
@@ -5,27 +5,52 @@ import MenuItem from '@mui/material/MenuItem';
 import MoreVertIcon from '@mui/icons-material/MoreVert';
 import { useState } from 'react';
 import { DeleteUserModal } from './DeleteUserModal.jsx';
+import { useNavigate } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectUserToEdit } from '../../store/usersFeature/usersSelectors.js';
+import { setUserToEdit } from '../../store/usersFeature/usersSlice.js';
 
-const options = ['Edit', 'Delete'];
+const options = ['Edit', 'Delete', 'Details'];
```
Comment (kzngoga) [inline src/components/users/UserActionMenu.jsx]: Change the menu item to **view details**

File: src/pages/UserDetailsPage.jsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            </CardContent>
+          </Card>
+        </Grid>
+        <Grid item xs={12} md={8}>
+          <Card
+            sx={{ height: '100%', display: 'flex', flexDirection: 'column' }}
+          >
+            <CardContent sx={{ flexGrow: 1 }}>
+              <Grid container spacing={2}>
+                {[
```
Comment (kzngoga) [inline src/pages/UserDetailsPage.jsx]: Declare this array outside the return function

**Review by kzngoga — CHANGES_REQUESTED** [review]
@MUTABAZI-D looks good, but I need you to work on some changes!

- This page needs to have a sidebar and navbar as the other pages. Pls also add a back button at the top to redirect to the **users page**
https://prnt.sc/5HkNaqoGrVy7

- Remove the **follow** & **message** buttons

**Comment by MUTABAZI-D** [comment]
hey boss @kzngoga , worked really carefully on requested changes !!! please review .

**Review by kzngoga — APPROVED** [review]
Looks good, now we can merge!

**Pull request merged**